### PR TITLE
feat: implement get token from service account

### PR DESCRIPTION
**Pull request by arielsepton**
This PR introduces the route: `GET /v1/namespaces/{namespace}/serviceaccounts/{serviceAccountName}/token`. 
This endpoint allows users to retrieve the token associated with a specified service account.

### CODE REVIEW

File: src/controllers/serviceaccounts.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+// serviceAccount implements the ServiceAccountController interface.
+type serviceAccountController struct {
+	client kubernetes.Interface
+	ctx    context.Context
+	logger *zap.Logger
+}
+
+// ServiceAccountController defines methods to interact with service accounts.
```
Comment (dvirgilad) [inline src/controllers/serviceaccounts.go:39]:
```suggestion
// ServiceAccountController defines methods to interact with ServiceAccounts.
```

File: src/controllers/serviceaccounts_test.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+			},
+			want: want{
+				serviceAccount: &corev1.ServiceAccount{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      testutils.ServiceAccountName,
+						Namespace: namespaceName,
+					},
+					Secrets: []corev1.ObjectReference{
+						{
+							Kind:      "Secret",
```
Comment (dvirgilad) [inline src/controllers/serviceaccounts_test.go:42]: Move to consts

File: src/routes/v1/serviceaccounts_test.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	}{
+		"ShouldSucceedGettingToken": {
+			args: args{
+				namespace:          testNamespaceName,
+				serviceAccountName: testutils.ServiceAccountName,
+			},
+			want: want{
+				statusCode: http.StatusOK,
+				response: map[string]interface{}{
+					testutils.TokenKey: "value",
```
Comment (dvirgilad) [inline src/routes/v1/serviceaccounts_test.go:46]: move to consts

File: src/controllers/serviceaccounts.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	tokenKey = "token"
+)
+
+const (
+	ErrCouldNotGetServiceAccount = "Could not get service account %q in namespace %q"
```
Comment (dvirgilad) [inline src/controllers/serviceaccounts.go:19]: ServiceAccount

File: src/routes/v1/serviceaccounts_test.go [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+			args: args{
+				namespace:          testNamespaceName + testutils.NonExistentSuffix,
+				serviceAccountName: testutils.ServiceAccountName,
+			},
+			want: want{
+				statusCode: http.StatusNotFound,
+				response: map[string]interface{}{
+					testutils.ErrorKey: fmt.Sprintf("%s, %s",
+						fmt.Sprintf(controllers.ErrCouldNotGetServiceAccount, testutils.ServiceAccountName, testNamespaceName+testutils.NonExistentSuffix),
+						fmt.Sprintf("serviceaccounts %q not found", testutils.ServiceAccountName),
```
Comment (dvirgilad) [inline src/routes/v1/serviceaccounts_test.go:60]: serviceaccount

File: src/routes/v1/serviceaccounts_test.go [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+			args: args{
+				namespace:          testNamespaceName,
+				serviceAccountName: testutils.ServiceAccountName + testutils.NonExistentSuffix,
+			},
+			want: want{
+				statusCode: http.StatusNotFound,
+				response: map[string]interface{}{
+					testutils.ErrorKey: fmt.Sprintf("%s, %s",
+						fmt.Sprintf(controllers.ErrCouldNotGetServiceAccount, testutils.ServiceAccountName+testutils.NonExistentSuffix, testNamespaceName),
+						fmt.Sprintf("serviceaccounts %q not found", testutils.ServiceAccountName+testutils.NonExistentSuffix),
```
Comment (dvirgilad) [inline src/routes/v1/serviceaccounts_test.go:76]: serviceaccount

File: src/utils/testutils/mocks/secret.go [hunk 7]
```diff
@@ -22,6 +22,41 @@ func PrepareSecret(name, namespace, dataKey, dataValue string) corev1.Secret {
...
+			testutils.TokenKey: []byte(tokenValue),
+		},
+	}
+}
+
+// PrepareDockerConfigSecret returns a mock docker config secret object.
+func PrepareDockerConfigSecret(name, namespace, serviceAccountTokenSecretName string) corev1.Secret {
+	ownerReference := metav1.OwnerReference{
+		APIVersion: "v1",
+		Kind:       "Secret",
```
Comment (dvirgilad) [inline src/utils/testutils/mocks/secret.go:46]: Can't this be taken from consts?

**Review by dvirgilad — COMMENTED** [review]
Great work!
Left some minor comments but overall looks great

File: src/routes/v1/serviceaccounts_test.go (hunk 6 above)
Comment (arielsepton) [reply src/routes/v1/serviceaccounts_test.go:76]: Same

**Review by dvirgilad — APPROVED** [review]
.

**Pull request merged**